### PR TITLE
Add warning about Autofac Update obsoletion

### PR DESCRIPTION
**Pull request by WilliamBZA**
Adds a note to explain that the container passed in to the `ExistingLifetimeScope` method will not be updated by any of the NServiceBus registrations.

Relates to https://github.com/Particular/NServiceBus.Autofac/issues/36

### CODE REVIEW

**Comment by WilliamBZA** [comment]
Ping @boblangley @distantcam @adamralph 

Please review.

**Comment by SimonCropp** [comment]
Given this is a breaking change I behaviour I assume it will be in a new major?

**Comment by indualagarsamy** [comment]
@WilliamBZA - Since the PlatDev link is not accessible to everyone, can you add enough context in the PR description, or link it to the appropriate issue in the NServiceBus.Autofac repo?

**Comment by WilliamBZA** [comment]
@SimonCropp This is not a breaking change. No API changes and no changes to the behaviour of the package.

@indualagarsamy It's adding details of already existing behaviour, it's actually irrelevant what the PlatDev issue contains as this should probably have been in before but we just hadn't noticed this behaviour until investigating.

File: nservicebus/containers/autofac.md [hunk 1]
```diff
@@ -20,6 +20,8 @@ snippet:Autofac
 
 snippet:Autofac_Existing
 
+NOTE: The registrations from the container  passed in to the `ExistingLifetimeScope` method will be copied into a new container. This means that the instance of the container passed in will not be altered and none of the internal NServiceBus registrations will appear in the original container.
```
Comment (adamralph) [inline nservicebus/containers/autofac.md]: Dunno if it matters, but there is an extra space between "container" and "passed".

Comment (WilliamBZA) [inline nservicebus/containers/autofac.md]: Good catch, updated.

File: nservicebus/containers/autofac.md [hunk 2]
```diff
@@ -20,6 +20,8 @@ snippet:Autofac
 
 snippet:Autofac_Existing
 
+NOTE: The registrations from the container passed in to the `ExistingLifetimeScope` method will be copied into a new container. This means that the instance of the container passed in will not be altered and none of the internal NServiceBus registrations will appear in the original container.
```
Comment (SimonCropp) [inline nservicebus/containers/autofac.md]: so this changes is being back ported to all versions?

**Comment by WilliamBZA** [comment]
Renamed back to WIP, the content will likely change again

File: nservicebus/containers/autofac.md (hunk 2 above)
Comment (WilliamBZA) [inline nservicebus/containers/autofac.md]: No. Good point. We're re-thinking our implementation

**Comment by boblangley** [comment]
I updated the content to remove the note about copying of the container and adding a warning about Autofac `ContainerBuilder.Update` being marked as obsolete.

**Comment by boblangley** [comment]
For more information about our change in direction see https://github.com/Particular/PlatformDevelopment/issues/1091#issuecomment-277108934

File: nservicebus/containers/autofac.md [hunk 3]
```diff
@@ -20,6 +20,7 @@ snippet:Autofac
 
 snippet:Autofac_Existing
 
+WARN: As of Autofac 4.2.1 the `Update` method on `ContainerBuilder` is marked as obsolete. In the future updating the container passed to NServiceBus may not be possible. See [the Autofac project issue](https://github.com/autofac/Autofac/issues/811) for more information.
```
Comment (adamralph) [inline nservicebus/containers/autofac.md]: Needs a couple of commas added:

> As of Autofac 4.2.1, the...

> In the future, updating...

**Comment by indualagarsamy** [comment]
@boblangley @WilliamBZA - As PlatDev is a private repo and any links you reference will not be available to the people that's outside of that domain as docs is a public repo, do you think you can add  NServiceBus.Autofac - issues#36 in the issue in the description?

**Pull request merged**